### ISSUE TITLE
MXNet 1.2 Test Accuracy Drop

### ISSUE THREAD

**Issue by haroldyang977** [apache/incubator-mxnet#11663]
Note: Providing complete information in the most concise form is the best way to get help. This issue template serves as the checklist for essential information to most of the technical issues and bug reports. For non-technical issues and feature requests, feel free to present the information in what you believe is the best form.

For Q & A and discussion, please start a discussion thread at https://discuss.mxnet.io 

## Description
After updating from MXNet 1.1 to 1.2, using the same code and configuration to re-train the model,  the accuracy drop 3~5%. I've re-trained my model multiple times, but MXNet 1.2 version always get lower accuracy. (I didn't encounter this when updating from 1.0 to 1.1)

## Environment info (Required)
Ubuntu 16.04
GPU: Nvidia GTX 1060

```
What to do:
----------Python Info----------
('Version      :', '2.7.12')
('Compiler     :', 'GCC 5.4.0 20160609')
('Build        :', ('default', 'Dec  4 2017 14:50:18'))
('Arch         :', ('64bit', 'ELF'))
------------Pip Info-----------
('Version      :', '9.0.3')
('Directory    :', '/usr/local/lib/python2.7/dist-packages/pip')
----------MXNet Info-----------
No MXNet installed.
----------System Info----------
('Platform     :', 'Linux-4.13.0-45-generic-x86_64-with-Ubuntu-16.04-xenial')
('system       :', 'Linux')
('node         :', 'TV01011569UP')
('release      :', '4.13.0-45-generic')
('version      :', '#50~16.04.1-Ubuntu SMP Wed May 30 11:18:27 UTC 2018')
----------Hardware Info----------
('machine      :', 'x86_64')
('processor    :', 'x86_64')
Architecture:          x86_64
CPU op-mode(s):        32-bit, 64-bit
Byte Order:            Little Endian
CPU(s):                8
On-line CPU(s) list:   0-7
Thread(s) per core:    2
Core(s) per socket:    4
Socket(s):             1
NUMA node(s):          1
Vendor ID:             GenuineIntel
CPU family:            6
Model:                 158
Model name:            Intel(R) Core(TM) i7-7700 CPU @ 3.60GHz
Stepping:              9
CPU MHz:               4054.336
CPU max MHz:           4200.0000
CPU min MHz:           800.0000
BogoMIPS:              7200.00
Virtualization:        VT-x
L1d cache:             32K
L1i cache:             32K
L2 cache:              256K
L3 cache:              8192K
NUMA node0 CPU(s):     0-7
Flags:                 fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf tsc_known_freq pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault invpcid_single pti retpoline intel_pt rsb_ctxsw spec_ctrl tpr_shadow vnmi flexpriority ept vpid fsgsbase tsc_adjust bmi1 hle avx2 smep bmi2 erms invpcid rtm mpx rdseed adx smap clflushopt xsaveopt xsavec xgetbv1 xsaves dtherm ida arat pln pts hwp hwp_notify hwp_act_window hwp_epp
----------Network Test----------
Setting timeout: 10
Timing for MXNet: https://github.com/apache/incubator-mxnet, DNS: 0.0156 sec, LOAD: 1.4830 sec.
Timing for PYPI: https://pypi.python.org/pypi/pip, DNS: 0.0714 sec, LOAD: 1.1005 sec.
Timing for FashionMNIST: https://apache-mxnet.s3-accelerate.dualstack.amazonaws.com/gluon/dataset/fashion-mnist/train-labels-idx1-ubyte.gz, DNS: 5.0151 sec, LOAD: 0.9209 sec.
Timing for Conda: https://repo.continuum.io/pkgs/free/, DNS: 0.2231 sec, LOAD: 0.0563 sec.
Timing for Gluon Tutorial(en): http://gluon.mxnet.io, DNS: 0.0567 sec, LOAD: 0.6574 sec.
Timing for Gluon Tutorial(cn): https://zh.gluon.ai, DNS: 0.5852 sec, LOAD: 0.7436 sec.

```

Package used (Python/R/Scala/Julia):
Python

## Error Message:
(Paste the complete error message, including stack trace.)
No Error

## Minimum reproducible example
(If you are using your own code, please provide a short script that reproduces the error. Otherwise, please provide link to the existing example.)
My training code is mainly based on following github site:
https://github.com/itijyou/ademxapp/blob/master/issegm/voc.py

## Steps to reproduce
(Paste the commands you ran that produced the error.)
My training code is mainly based on the above github site. The mainly difference are I trained on my own model and dataset I can't provide here.
My model use bilinear initializer and deconvolution for bilinear-upsampling.
The weights initializer is Xavier, and I use SGD as optimizer. All the same as the provided github site.

## What have you tried to solve it?
1. The inference result between MXNet 1.1 and 1.2 are the same. So I think restoring the checkpoint weights to the symbols is OK.

Can someone give me a hint or direction on how to investigate on this issue?

**Comment by apeforest** [comment]
@sandeep-krishnamurthy Please label this issue _Need Triage_. Thanks!

**Comment by haroldyang977** [comment]
Sorry for bothering.
After I re-install the MXNet by pip as usual, and update to MXNet 1.2.1. Somehow the problem seems solved.
